src/components/source_editor/LatticeConfigurationDialog.js

```diff
@@ -108,6 +111,22 @@ class LatticeConfigurationDialog extends React.Component {
         const newMaterial = new Material(newMaterialConfig);
         // assert basis is stored in 'crystal' units
         newMaterial.toCrystal();
+
+        // update to non-periodic if asked to do so
+        if (this.state.isNonPeriodic) {
+            const newBasis = new Made.Basis({
```

Comment:
Why not just reuse the newMaterial's Basis directly??

Reply:
touche, will fix,